pages/index.jsx

```diff
@@ -5,12 +5,25 @@ import {auth,firebase} from '../lib/firebase';
 const index = () => {
   console.log(auth);
   console.log(firebase);
+  const provider = new firebase.auth.GoogleAuthProvider();
+  const loginHandler = ()=>{
+    firebase.auth().signInWithPopup(provider)
+    .then(()=>{
+      const token = result.credential.accessToken;
+      const user = result.user;
+    }).catch((error)=>{
+      const errorCode = error.code;
+      const errorMessage = error.message;
+      const email = error.email;
+      const credential = error.credential;
+      });
+  }
   return (
     <App>
       <div>
         <h1>Index here bruh</h1>
         <Link href="/game">
```

Comment:
This is a wrong way. Call the push('/game') when your auth promise is resolved in then().
> Resource to inject router properties into a page : [Link](https://github.com/zeit/next.js/blob/canary/examples/dynamic-routing/pages/post/%5Bid%5D/index.js)